SPECS/python-pip/CVE-2025-50181.patch

```diff
@@ -0,0 +1,61 @@
+From d75419bb31f1e5527773d1e22586881241c2f774 Mon Sep 17 00:00:00 2001
+From: jykanase <v-jykanase@microsoft.com>
+Date: Mon, 23 Jun 2025 13:00:58 +0000
+Subject: [PATCH] CVE-2025-50181
+
+Upstream Patch Reference: https://github.com/urllib3/urllib3/commit/f05b1329126d5be6de501f9d1e3e36738bc08857
+---
+ src/pip/_vendor/urllib3/poolmanager.py | 29 +++++++++++++++++++++++---
+ 1 file changed, 26 insertions(+), 3 deletions(-)
+
+diff --git a/src/pip/_vendor/urllib3/poolmanager.py b/src/pip/_vendor/urllib3/poolmanager.py
+index fb51bf7..d654acd 100644
+--- a/src/pip/_vendor/urllib3/poolmanager.py
++++ b/src/pip/_vendor/urllib3/poolmanager.py
+@@ -168,9 +168,32 @@ class PoolManager(RequestMethods):
+     proxy = None
+     proxy_config = None
+ 
+-    def __init__(self, num_pools=10, headers=None, **connection_pool_kw):
+-        RequestMethods.__init__(self, headers)
++    def __init__(
++        self,
++        num_pools: int = 10,
++        headers: typing.Mapping[str, str] | None = None,
```

Comment:
I think we should import the typing module in this specific file.
Since Python is an interpreted language - this would break at runtime whenever the code path flows through this and given that the file that is patched, does not import the typing module, it will break.